reframe/core/schedulers/slurm.py

```diff
@@ -293,11 +293,24 @@ def _check_and_cancel(self, reason_descr):
             reason, reason_details = reason_descr, None
 
         if reason in self._cancel_reasons:
-            # Here we handle the case were the UnavailableNodes list is empty,
-            # which actually means that the job is pending
             if reason == 'ReqNodeNotAvail' and reason_details:
-                if re.match(r'UnavailableNodes:$', reason_details.strip()):
-                    return
+                nodes_match = re.match(
+                    r'UnavailableNodes:(?P<node_names>\S+)?',
+                    reason_details.strip())
+                if nodes_match:
+                    node_names = node_match.group('node_names')
```

Comment:
I think the following will work:

```suggestion
                    node_names = node_match['node_names']
```